sentry/src/main/java/io/sentry/connection/UdpConnection.java

```diff
@@ -0,0 +1,187 @@
+     */
+    private final Set<EventSendCallback> eventSendCallbacks;
+    /**
+     * JsonFactory for creating chunks.
+     */
+    private final JsonFactory jsonFactory;
+    /**
+     * The maximum chunk size beyond which a new UDP message will be sent.
+     */
+    private int maxChunkSize;
+    /**
+     * InetAddress for the remote UDP socket.
+     */
+    private InetAddress address = null;
+    /**
+     * UDP Socket.
+     */
+    private DatagramSocket socket = null;
+
+
+    /**
+     * Create a UDP connection.
+     * @param dsn for example. udp://public:private@localhost:1234/1.
+     * @param maxChunkSize maxChunkSize.
+     */
+    public UdpConnection(Dsn dsn, int maxChunkSize) {
+        this.host = dsn.getHost();
+        this.port = dsn.getPort();
+        this.maxChunkSize = maxChunkSize;
+        this.jsonFactory = new JsonFactory();
+        this.marshaller = new JsonMarshaller();
+        this.marshaller.setCompression(false);
+        this.eventSendCallbacks = new HashSet<>();
+    }
+
+
+    private void init() throws UnknownHostException, SocketException {
+        if (address == null) {
+            address = InetAddress.getByName(this.host);
+        }
+        if (socket == null) {
+            socket = new DatagramSocket();
+        }
+    }
+
+    @Override
+    public void send(Event event) throws ConnectionException {
+        try {
+
+            this.init();
+
+            ByteArrayOutputStream destination = new ByteArrayOutputStream();
+            marshaller.marshall(event, destination);
+            byte[] eventBuf = destination.toByteArray();
+
+            List<String> chunks = new ArrayList<>();
+            chunkify(new String(eventBuf), chunks);
+            int totalChunks = chunks.size();
+            for (int numChunk = 0; numChunk < totalChunks; numChunk++) {
+                String chunk = chunks.get(numChunk);
+                String eventId = event.getId().toString();
+                byte[] sendArray = this.getChunkBytes(chunk, eventId, numChunk + 1, totalChunks);
+                DatagramPacket packet = new DatagramPacket(sendArray, sendArray.length, address, this.port);
+                this.sendPacket(packet);
+            }
+
+            for (EventSendCallback eventSendCallback : eventSendCallbacks) {
+                eventSendCallback.onSuccess(event);
+            }
+
+        } catch (IOException ioex) {
+            for (EventSendCallback eventSendCallback : eventSendCallbacks) {
+                eventSendCallback.onFailure(event, ioex);
+            }
+            throw new ConnectionException("Error writing to udp", ioex);
+        }
+    }
+
+    /**
+     * Package private method to allow mocking of socket send.
+     * @param packet DataGram Packet to send.
+     * @throws IOException can be thrown if socket send fails.
+     */
+    void sendPacket(DatagramPacket packet) throws IOException {
+        this.socket.send(packet);
```

Comment:
Is this threadsafe? Connections are usually wrapped in an `AsyncConnection` that can call `send` concurrently. Since it's UDP and there's no setup/teardown I wonder if we should just construct and destroy the socket per send?